storage/jellyfish_merkle/src/node_type/mod.rs

```diff
@@ -89,25 +89,26 @@ impl NodeKey {
     pub fn encode(&self) -> Result<Vec<u8>> {
         let mut out = vec![];
         out.write_u64::<BigEndian>(self.version())?;
-        out.write_u8((self.nibble_path().num_nibbles() & 1) as u8)?;
         out.write_all(self.nibble_path().bytes())?;
+        if self.nibble_path().num_nibbles() != 0 && self.nibble_path().num_nibbles() % 2 == 0 {
```

Comment:
It might be worth writing down the encoding rules as comments.

Reply:
Is it necessary to special case the empty nibble path?

Reply:
I think it is desirable to have just the version as the key for the root node.